tests/test_misc.py

```diff
@@ -1221,3 +1221,70 @@ def test_const_dict_variable_python_type(self):
             ConstDictVariable(d2, collections.OrderedDict).python_type(),
             collections.OrderedDict,
         )
+
+    def test_builtin_subclasses_as_method_on_class_type(self):
+        class Foo:
+            def __init__(name):
+                name_ = name
+            
+            def get_name(self):
+                return "Foo " + self.name_
+
+        class Bar(Foo):
+            def __init__(name):
+                name_ = name
+            
+            def get_name(self):
+                return "Bar " + self.name_
+
+        class Baz(Foo):
+            def __init__(name):
+                name_ = name
+            
+            def get_name(self):
+                return "Baz " + self.name_
+
+        subs_of_foo_reg = Foo.__subclasses__()
+
+        subs_of_foo_optim = list()
+        counter = CompileCounter()
+        with torchdynamo.optimize_assert(counter):
+            subs_of_foo_optim = Foo.__subclasses__()
```

Comment:
This won't actually get captured by TorchDynamo because it isn't in a new frame.  Perhaps use dectorator syntax:
```
@torchdynamo.optimize_assert(counter)
def fn():
  return Foo.__subclasses__()
```